pkg/base/replicated.go

```diff
@@ -152,7 +157,149 @@ func renderReplicated(u *upstreamtypes.Upstream, renderOptions *RenderOptions) (
 		}
 	}
 
-	return &commonBase, helmBases, nil
+	return &commonBase, helmBases, kotsKindsFileMap, nil
+}
+
+func renderKotsKinds(upstreamFiles []upstreamtypes.UpstreamFile, renderedConfig *kotsv1beta1.Config, renderOptions *RenderOptions, builder *template.Builder) (*kotsutil.KotsKinds, map[string][]byte, error) {
+	fileContentsMap := make(map[string][]byte)
+	kotsKinds := kotsutil.EmptyKotsKinds()
+
+	for _, upstreamFile := range upstreamFiles {
+		document := &Document{}
+		if err := yaml.Unmarshal(upstreamFile.Content, document); err != nil {
+			continue
+		}
+
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		decoded, gvk, err := decode(upstreamFile.Content, nil, nil)
+		if err != nil {
+			if document.APIVersion == "kots.io/v1beta1" || document.APIVersion == "troubleshoot.sh/v1beta2" || document.APIVersion == "velero.io/v1" {
+				return nil, nil, errors.Wrapf(err, "failed to decode kots yaml %s", upstreamFile.Path)
+			}
+			continue
+		}
+
+		switch gvk.String() {
+		case "kots.io/v1beta1, Kind=Config":
+			// Use the rendered Config instead of the upstream.
+			kotsKinds.Config = renderedConfig
+			configBytes, err := yaml.Marshal(renderedConfig)
+			if err != nil {
+				return nil, nil, errors.Wrap(err, "failed to marshal rendered config")
+			}
+			fileContentsMap[upstreamFile.Path] = configBytes
+
+		case "kots.io/v1beta1, Kind=ConfigValues":
+			fileContentsMap["configvalues.yaml"] = upstreamFile.Content
+			kotsKinds.ConfigValues = decoded.(*kotsv1beta1.ConfigValues)
+
+		case "kots.io/v1beta1, Kind=HelmChart":
```

Comment:
we shouldn't do any processing for this custom resource. it should just be the same as the others. the goal is to have visibility into the rendered version of the custom resource, and this does more than that. what if we want to check if `optionalValues` rendered correctly?